src/controller/UserController.ts

```diff
@@ -0,0 +1,90 @@
+import { Request, Response } from "express";
+import { UserInputDTO} from "../types/userInputDTO";
+import { UserBusiness } from "../business/UserBusiness";
+import { BaseDatabase } from "../data/BaseDatabase";
+import { BandInputDTO } from "../model/Band";
+import { loginInputDTO } from "../types/loginInputDTO";
+
+
+export class UserController {
+    constructor(
+        private userBusiness: UserBusiness
+
+    ){};
+
+    async signup(request: Request, response: Response) {
+        try {
+            const input: UserInputDTO = {
+                email: request.body.email,
+                name: request.body.name,
+                password: request.body.password,
+                role: request.body.role
+            }
+            
+            
+            const token = await this.userBusiness.signUp(input);
+
```

Comment:
Quando usamos essa sintaxe de função com ts, ele não reconhece o this, isso significa que quando o código chega na linha 25, ele não funciona pois this seria um valor undefined, a solução seria usar uma arrow function, dessa forma ficaria:

signup =  async (request: Request, response: Response)) => {...}

A mesmo coisa vale para login